cli/internal/cmd/upgradeapply.go

```diff
@@ -172,6 +183,20 @@ func (u *upgradeApplyCmd) upgradeApply(cmd *cobra.Command, fileHandler file.Hand
 	return nil
 }
 
+func diffAttestationCfg(currentAttestationCfg config.AttestationCfg, newAttestationCfg config.AttestationCfg) (string, error) {
+	// cannot compare structs directly with go-cmp because of unexported fields in the attestation config
+	currentYml, err := yaml.Marshal(currentAttestationCfg)
+	if err != nil {
+		return "", fmt.Errorf("marshalling remote attestation config: %w", err)
+	}
+	newYml, err := yaml.Marshal(newAttestationCfg)
+	if err != nil {
+		return "", fmt.Errorf("marshalling local attestation config: %w", err)
+	}
+	diff := string(diff.Diff("current", currentYml, "new", newYml))
+	return diff, nil
+}
```

Comment:
Feel free to ignore, but we may want to only use one diffing library and we already use hexops/gotextdiff in other places.

Example:
https://github.com/edgelesssys/constellation/blob/cc106132520202a225821bfa8b7f2e33891b27a9/hack/bazel-deps-mirror/internal/bazelfiles/files.go#L122-L131

Reply:
We were discussing about it with Paul and he mentioned the library too, but decided against it cause it has no benefit, harder usability,  and the library was last updated 3 years ago. Since we already use `go-internal` anyways we decided for that one.